tests/components/reolink/test_number.py

```diff
@@ -0,0 +1,114 @@
+"""Test the Reolink number platform."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from reolink_aio.api import Chime
+from reolink_aio.exceptions import InvalidParameterError, ReolinkError
+
+from homeassistant.components.number import (
+    ATTR_VALUE,
+    DOMAIN as NUMBER_DOMAIN,
+    SERVICE_SET_VALUE,
+)
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .conftest import TEST_NVR_NAME
+
+from tests.common import MockConfigEntry
+
+
+async def test_number(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    reolink_connect: MagicMock,
+) -> None:
+    """Test number entity with volume."""
+    reolink_connect.volume.return_value = 80
+
+    with patch("homeassistant.components.reolink.PLATFORMS", [Platform.NUMBER]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    entity_id = f"{Platform.NUMBER}.{TEST_NVR_NAME}_volume"
+
+    assert hass.states.get(entity_id).state == "80"
+
+    reolink_connect.set_volume = AsyncMock()
```

Comment:
Is reolink_connect an AsyncMock as well? I think we can omit this step

Reply:
Indeed you are right, this can be ommited.
Removed it throught all reolink tests.